src/ansys/geometry/core/_grpc/_services/v1/beams.py

```diff
@@ -43,26 +54,201 @@ class GRPCBeamsServiceV1(GRPCBeamsService):
 
     @protect_grpc
     def __init__(self, channel: grpc.Channel):  # noqa: D102
-        from ansys.api.discovery.v1.assignments.beams import BeamsStub
+        from ansys.api.discovery.v1.design.extensions.beam_pb2_grpc import (
+            BeamStub,
+        )
+        from ansys.api.discovery.v1.engineeringdata.beamprofiledata_pb2_grpc import (
+            BeamProfileDataStub,
+        )
 
-        self.stub = BeamsStub(channel)
+        self.beam_profile_commands_stub = BeamProfileDataStub(channel)
+        self.beam_commands_stub = BeamStub(channel)
 
     @protect_grpc
     def create_beam_segments(self, **kwargs) -> dict:  # noqa: D102
-        raise NotImplementedError
+        from ansys.api.discovery.v1.commonmessages_pb2 import Line
+        from ansys.api.discovery.v1.engineeringdata.beamprofiledata_pb2 import (
+            CreateBeamProfileSegmentsRequest,
+            CreateBeamProfileSegmentsRequestData,
+        )
+
+        # Create the gRPC Line objects
+        lines = []
+        for segment in kwargs["segments"]:
+            lines.append(
+                from_line_to_grpc_line(
+                    Line(
+                        start=from_point3d_to_grpc_point(segment[0]),
+                        end=from_point3d_to_grpc_point(segment[1]),
+                    )
+                )
+            )
```

Comment:
Same comment... I'm not sure how this is even working. You shouldn't be needing to pass through the ``from_line_to_grpc_line`` method.